atlite/data.py

```diff
@@ -124,28 +128,29 @@ def cutout_prepare(cutout, features=slice(None), tmpdir=None, overwrite=False):
         logger.info(f'Calculating and writing with module {module}:')
         missing_features = missing_vars.index.unique('feature')
         ds = get_features(cutout, module, missing_features, tmpdir=tmpdir)
-        # make sure we don't loose any unused dimension by selecting
-        ds = ds[missing_vars.values].assign_coords(ds.coords)
-
-        ds = ds.assign_attrs(**cutout.data.attrs)
         prepared |= set(missing_features)
-        ds = ds.assign_attrs(prepared_features = list(prepared))
-        # convert bool to int for netCDF4 storing
-        ds.attrs.update({k: v if not isinstance(v, bool) else int(v)
-                         for k,v in ds.attrs.items()})
 
+        ds = cutout.data.merge(ds[missing_vars.values])\
+                            .assign_attrs(**cutout.data.attrs, **ds.attrs)
+        ds.attrs = non_bool_dict(cutout.data.attrs)
+        ds.attrs.update({'prepared_features': list(prepared)})
+
+        # write data to tmp file, copy it to original data, this is much saver
+        # than appending variables
+        tstr = next(tmpname())
+        tmp = cutout.path.parent / (f'tmp_{tstr}' + cutout.path.stem + '.nc')
         with ProgressBar():
-            if cutout.path.exists():
-                mode = 'a'
-            else:
-                mode = 'w'
+            ds.to_netcdf(tmp)
+            ds.close()
+        cutout.data.close()
```

Comment:
If this is a user provided dataset, i don't think we should close it. That's what the `is_view` attribute was for.